### DISCUSSION TITLE
Convert your async into sync - sync method source generator

### DISCUSSION THREAD

**Discussion by virzak** [dotnet/csharplang#6596] [Show and tell]
Hi all,

If anyone needs to have both sync and async versions of a method, you can use [sync method generator](https://github.com/zompinc/sync-method-generator).

The generator uses [CSharpSyntaxRewriter](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.csharpsyntaxrewriter) to strip or replace the code of async/await, CancellationToken, IProgress, GetAsyncEnumerable, Task, Memory\<T>.

Hoping to get your input and suggestions.

**Comment by CyrusNajmabadi** [comment]
Hi @virzak this isn't really the right location for such a discussion.  Maybe try the .net or c# discords?  like discord.gg/csharp?
